code/modules/clothing/glasses/glasses.dm

```diff
@@ -257,6 +257,9 @@
 		if(noir_mode)
 			if(color_view && user.client && !user.client.color)
 				animate(user.client, color = color_view, time = 10)
+	else
+		if(user.client && user.client.color)

```

Comment:
For whatever reason, instead of setting user.client.color to color_view (16 entries in the list), it sets user.client.color to color_view + 4 extra entries with a value of 0 (20 entries total, the extra ones all trailing zeroes). Wierd!